src/llrb-tree/LlrbTree.ts

```diff
@@ -175,6 +175,151 @@ export class LlrbTree<K, V> implements SonicMap<K, V, LlrbNode<K, V>> {
     }
   }
 
+  // Helper methods for LLRB deletion
+  private _isRed(node: LlrbNode<K, V> | undefined): boolean {
+    return node ? !node.b : false;
+  }
+
+  private _rotateLeft(node: LlrbNode<K, V>): LlrbNode<K, V> {
+    const x = node.r!;
+    node.r = x.l;
+    if (x.l) x.l.p = node;
+    x.l = node;
+    x.p = node.p;
+    node.p = x;
+    
+    // Update parent's child pointer
+    if (x.p) {
+      if (x.p.l === node) x.p.l = x;
+      else x.p.r = x;
+    } else {
+      this.root = x;
+    }
+    
+    x.b = node.b;
+    node.b = false;
+    return x;
+  }
+
+  private _rotateRight(node: LlrbNode<K, V>): LlrbNode<K, V> {
```

Comment:
Extract simple helper functions which you can, like `_rotateRight` etc., into a standalone `util.ts` module.